src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -72,10 +68,80 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) throws IOExceptio
         attributes.put("X-Amz-Expires", "30");
     }
 
+    DescribeInstances(AwsConfig awsConfig) {
+        this.awsConfig = awsConfig;
+    }
+
+    void checkKeysFromIamRoles(Environment env) throws IOException {
+        if (awsConfig.getAccessKey() != null) {
+            return;
+        }
+
+        if (awsConfig.getIamRole() == null) {

```

Comment:
@mesutcelik I think we should keep the non-existence of the <iam-role> as a non-valid case, when not running in ECS (and when no access/secret key has been defined).